### PR TITLE
added sign message to other methods and validated the base64 signed m…

### PR DESCRIPTION
**Pull request by ahujsak**
…essages for segwit and non-segwit

### CODE REVIEW

File: src/engine/currencyEngine.js [hunk 1]
```diff
@@ -125,7 +122,7 @@ export class CurrencyEngine {
   }: CurrencyEngineSettings) {
     // Validate that we are a valid EdgeCurrencyEngine:
     // eslint-disable-next-line no-unused-vars
-    const test: EdgeCurrencyEngine = this
```
Comment (swansontec) [inline src/engine/currencyEngine.js]: This line needs to be present. It helps Flow give us better errors.

File: src/engine/currencyEngine.js [hunk 2]
```diff
@@ -3,7 +3,6 @@
 import { bns } from 'biggystring'
 import { type Disklet } from 'disklet'
 import {
-  type EdgeCurrencyEngine,
```
Comment (swansontec) [inline src/engine/currencyEngine.js]: Restore this line as well.

File: src/engine/currencyEngine.js [hunk 3]
```diff
@@ -626,21 +634,9 @@ export class CurrencyEngine {
       throw e
     }
   }
-
```
Comment (swansontec) [inline src/engine/currencyEngine.js]: Leave the blank lines between methods.

File: src/engine/keyManager.js [hunk 4]
```diff
@@ -290,32 +290,7 @@ export class KeyManager {
   }
 
   async signMessage ({ message, address }: SignMessage) {
```
Comment (swansontec) [inline src/engine/keyManager.js]: Is the `signMessage` method even used anymore? Can we remove the whole thing?

File: yarn.lock [hunk 5]
```diff
@@ -2288,6 +2295,19 @@ elliptic@^6.0.0, elliptic@^6.2.3, elliptic@^6.4.0:
     minimalistic-assert "^1.0.0"
     minimalistic-crypto-utils "^1.0.0"
 
+elliptic@^6.4.1:
```
Comment (swansontec) [inline yarn.lock]: We should delete all the `elliptic` entries in yarn.lock, then re-run yarn, to consolidate versions.

File: src/utils/utils.js [hunk 6]
```diff
@@ -32,6 +33,11 @@ export async function hash256 (hex: any) {
   return Promise.resolve(hash256Sync(hex))
 }
 
+export async function secp256k1Sign (message: Buffer, key: Buffer) {
+  const sigObj = await secp256k1.sign(message, key)
```
Comment (swansontec) [inline src/utils/utils.js]: Talk to Eliran about nonces. Doesn't this use https://tools.ietf.org/html/rfc6979 already, so we don't need to do anyting?

File: src/engine/keyManager.js (hunk 4 above)
Comment (thehobbit85) [reply src/engine/keyManager.js]: I think someone is using it.
I'm not 100% but I do remember we created it for a reason.
Keep it here for now and mention it at the next engineering meeting.

File: src/utils/utils.js (hunk 6 above)
Comment (swansontec) [reply src/utils/utils.js]: Talked to Eliran about this, and it sounds like we are doing the nonces correctly.

Comment (thehobbit85) [reply src/utils/utils.js]: This is verified to be using rfc6979, so no need to change this.

**Pull request merged**